app/src/main/java/com/microsoft/device/samples/dualscreenexperience/presentation/history/ui/OrderDetails.kt

```diff
@@ -0,0 +1,218 @@
+    bottomNavPadding: Int,
+    isLandscape: Boolean
+) {
+    if (order == null || showTwoPages == null) {
+        // REVISIT
+        return
+    }
+
+    // Calculate padding for LazyColumn
+    val paddingValues = with(LocalDensity.current) {
+        PaddingValues(bottom = 20.dp + topBarPadding.toDp() + bottomNavPadding.toDp())
+    }
+
+    val columnModifier = if (isLandscape)
+        Modifier
+            .fillMaxWidth(0.915f)
+            .padding(top = 32.dp)
+            .fillMaxHeight()
+    else
+        Modifier
+            .fillMaxWidth(0.9f)
+            .fillMaxHeight()
+
+    Box(modifier = Modifier.fillMaxSize(), contentAlignment = Alignment.TopCenter) {
+        Column(
+            modifier = columnModifier,
+        ) {
+            OrderHeader(order, showTwoPages)
+            Spacer(modifier = Modifier.height(22.dp))
+            OrderItems(order.items, paddingValues)
+        }
+    }
+}
+
+@Composable
+fun OrderHeader(order: Order, showTwoPages: Boolean) {
+    if (showTwoPages) {
+        Text(
+            text = stringResource(R.string.toolbar_history_details_title),
+            style = MaterialTheme.typography.h4,
+            color = MaterialTheme.colors.onBackground
+        )
+        Spacer(Modifier.height(4.dp))
+    }
+
+    Row(modifier = Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween) {
+        OrderDate(orderTimestamp = order.orderTimestamp)
+        Column(verticalArrangement = spacedBy(4.dp), horizontalAlignment = Alignment.End) {
+            OrderId(orderId = order.orderId)
+            OrderAmount(orderPrice = order.totalPrice)
+        }
+    }
+}
+
+@Composable
+fun OrderItems(orderItems: MutableList<OrderItem>, paddingValues: PaddingValues) {
+    LazyColumn(
+        verticalArrangement = spacedBy(25.dp),
+        contentPadding = paddingValues
+    ) {
+        orderItems.map { orderItem ->
+            item {
+                OrderItemPreview(orderItem = orderItem)
+            }
+        }
+    }
+}
+
+@Composable
+fun OrderItemPreview(orderItem: OrderItem) {
+    Box {
+        OrderItemDetails(orderItem = orderItem)
+        OrderItemImage(orderItem = orderItem)
+    }
+}
+
+@Composable
+fun BoxScope.OrderItemDetails(orderItem: OrderItem) {
+    Row(
+        modifier = Modifier
+            .fillMaxSize()
+            .align(Alignment.BottomCenter),
+        horizontalArrangement = spacedBy(41.dp)
+    ) {
+        OrderItemBox()
+        OrderItemText(orderItem = orderItem)
+        OrderItemViewButton(orderItem = orderItem)
+    }
+}
+
+@Composable
+fun RowScope.OrderItemText(orderItem: OrderItem) {
+    Column(modifier = Modifier.weight(5f), verticalArrangement = spacedBy(6.dp)) {
+        Text(
+            text = orderItem.name,
+            style = MaterialTheme.typography.body2,
+            color = MaterialTheme.colors.onBackground
+        )
+        Text(
+            // REVISIT: extract common code from binding adapter formatPrice
```

Comment:
This formatting was copied from a [binding adapter](https://github.com/microsoft/surface-duo-dual-screen-experience-example/blob/7d98990d8ad1905a9cfc52532dc157555047d75e/app/src/main/java/com/microsoft/device/samples/dualscreenexperience/presentation/util/BindingAdapters.kt#L91) - would it make sense to extract the common code into a utility class/module? If so, where would be the best place to put it?